dbms/src/Client/Connection.h

```diff
@@ -57,7 +57,6 @@ class Connection : private boost::noncopyable
     Connection(const String & host_, UInt16 port_,
         const String & default_database_,
         const String & user_, const String & password_,
-        const ConnectionTimeouts & timeouts_,
```

Comment:
It's just a question. Why you don't want to save old behaviour? There are a lot of places in code, where fixed timeout is OK. Constructor may take `default_timeouts`, and in each function we can pass non default, otherwise they would use default. Diff would be much smaller, and `Connection` interface would be more convenient.

Reply:
This way it was easier to debug as missing argument or variable is a compilation error while wrong default value requires complicated integration tests. On the other hand I don't think the diff would be much smaller because of additional logic to chose between default value and provided via method argument.